src/Compilation.zig

```diff
@@ -699,6 +699,7 @@ pub const InitOptions = struct {
     rpath_list: []const []const u8 = &[0][]const u8{},
     c_source_files: []const CSourceFile = &[0]CSourceFile{},
     link_objects: []const []const u8 = &[0][]const u8{},
+    must_link_objects: []const []const u8 = &[0][]const u8{},
```

Comment:
There are a few places throughout the codebase that the `link_objects` array (1 line above) is used for various things; to calculate the root name, choose the linker backend, caching stuff, and probably more that I'm not thinking of. In order to avoid introducing subtle bugs, I suggest this change:

```diff
-    link_objects: []const []const u8 = &[0][]const u8{},
-    must_link_objects: []const []const u8 = &[0][]const u8{}, 
+    link_objects: []const LinkObject = &.{},
```

With `LinkObject` defined like this:

```zig
const LinkObject = struct {
    path: []const u8,
    must_link: bool = false,
};
```

You'll have to go and edit a bunch of places in the code to make this change but I think it will reveal problems with this patch, and make it straightforward to fix and maintain.

Reply:
Changes introduced in 59ef553 - have a look and lemme know if that's what you had in mind!